crates/sui/Cargo.toml

```diff
@@ -133,3 +133,6 @@ gas-profiler = [
     "sui-types/gas-profiler",
     "sui-execution/gas-profiler",
 ]
+# TODO remove once compatibility check is fully finished
+# not meant for long term use
+compatibility-check-errors = []
```

Comment:
To be honest, I would just pull the relevant changes out into their own commit so I could keep working on them locally, but land the rest of the stuff. Working on the formatting so we can enable this feature is going to be our immediate next step, so I wouldn't worry about checking in what you've already got (integration-wise) gated behind a feature.

Reply:
Yeah it just makes the git commands easier since I do like branching off of branches and continuing work. I have two branches off of compatibility-cli-mode (flags and now formatting) so it's preferable that I keep as much of the common in here.

The feature flag really is just to keep dead code warnings away.

Reply:
nevermind the flags liter a lot of the code, I'm going to just keep it out of the PR.